lib/rules/utils/ast-utils.js

```diff
@@ -37,7 +37,7 @@ const LINEBREAKS = new Set(["\r\n", "\r", "\n", "\u2028", "\u2029"]);
 // A set of node types that can contain a list of statements
 const STATEMENT_LIST_PARENTS = new Set(["Program", "BlockStatement", "SwitchCase"]);
 
-const DECIMAL_INTEGER_PATTERN = /^(0|[1-9]\d*)$/u;
+const DECIMAL_INTEGER_PATTERN = /^(0|[1-9](?:_?\d)*)$/u;
```

Comment:
This actually had a bug: it doesn't recognize [NonOctalDecimalIntegerLiteral](https://www.ecma-international.org/ecma-262/11.0/index.html#prod-annexB-NonOctalDecimalIntegerLiteral) literals, like `08`.

I think we can fix this later since it's an existing edge-case bug that isn't related to numeric literals and requires a non-trivial change.

Reply:
Mind making an issue?

Reply:
Opened https://github.com/eslint/eslint/issues/13588